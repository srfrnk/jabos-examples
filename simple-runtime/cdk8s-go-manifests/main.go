package main

import (
	"fmt"
	"os"

	"example.com/cdk8s-go-manifests/imports/k8s"
	"github.com/aws/constructs-go/constructs/v3"
	"github.com/aws/jsii-runtime-go"
	"github.com/cdk8s-team/cdk8s-core-go/cdk8s"
)

type MyChartProps struct {
	cdk8s.ChartProps
}

func NewMyChart(scope constructs.Construct, id string, props *MyChartProps) cdk8s.Chart {
	var cprops cdk8s.ChartProps
	if props != nil {
		cprops = props.ChartProps
	}
	chart := cdk8s.NewChart(scope, jsii.String(id), &cprops)

	label := map[string]*string{"app": jsii.String("test-deployment-cdk8s-go")}

	k8s.NewKubeDeployment(chart, jsii.String("deployment"), &k8s.KubeDeploymentProps{
		Metadata: &k8s.ObjectMeta{
			Name:   jsii.String("test-deployment-cdk8s-go"),
			Labels: &label,
		},
		Spec: &k8s.DeploymentSpec{
			Replicas: jsii.Number(1),
			Selector: &k8s.LabelSelector{
				MatchLabels: &label,
			},
			Template: &k8s.PodTemplateSpec{
				Metadata: &k8s.ObjectMeta{
					Labels: &label,
				},
				Spec: &k8s.PodSpec{
					Containers: &[]*k8s.Container{{
						Name:  jsii.String("test-deployment-cdk8s-go"),
						Image: jsii.String(fmt.Sprintf("registry.kube-system:80/example-image-%s:%s", os.Getenv("BUILD_ENV"), os.Getenv("LATEST_COMMIT_ID"))),
					}},
				},
			},
		},
	})

	return chart
}

func main() {
	app := cdk8s.NewApp(nil)
	NewMyChart(app, "cdk8s-go-manifests", nil)
	app.Synth()
}
