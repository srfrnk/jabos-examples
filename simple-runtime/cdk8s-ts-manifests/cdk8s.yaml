language: typescript
app: npm run compile && node main.js
imports:
  - k8s
