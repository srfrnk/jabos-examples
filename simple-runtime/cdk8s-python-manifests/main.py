#!/usr/bin/env python
from constructs import Construct
from cdk8s import App, Chart
from imports import k8s
import os


class MyChart(Chart):
    def __init__(self, scope: Construct, id: str):
        super().__init__(scope, id)

        k8s.KubeDeployment(scope=self, id='deployment',
                           metadata=k8s.ObjectMeta(name="test-deployment-cdk8s-python", labels={"app": "test-deployment-cdk8s-python"}),
                           spec=k8s.DeploymentSpec(
                               replicas=1,
                               selector=k8s.LabelSelector(match_labels={"app": "test-deployment-cdk8s-python"}),
                               template=k8s.PodTemplateSpec(
                                   metadata=k8s.ObjectMeta(labels={"app": "test-deployment-cdk8s-python"}),
                                   spec=k8s.PodSpec(
                                       containers=[
                                           k8s.Container(
                                               image=f"registry.kube-system:80/example-image-{os.environ.get('BUILD_ENV')}:{os.environ.get('LATEST_COMMIT_ID')}",
                                               name="test-deployment-cdk8s-python",
                                           ),
                                       ]
                                   )
                               )
                           ))


app = App()
MyChart(app, "cdk8s-python-manifests")

app.synth()
