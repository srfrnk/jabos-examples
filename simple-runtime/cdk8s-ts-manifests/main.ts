import { Construct } from 'constructs';
import { App, Chart, ChartProps } from 'cdk8s';
import { KubeDeployment } from './imports/k8s';

export class MyChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);

    new KubeDeployment(this, 'deployment', {
      metadata: {
        labels: {
          app: "test-deployment-cdk8s"
        },
        name: "test-deployment-cdk8s"
      },
      spec: {
        replicas: 1,
        selector: {
          matchLabels: {
            app: "test-deployment-cdk8s"
          }
        },
        template: {
          metadata: {
            labels: {
              app: "test-deployment-cdk8s"
            }
          },
          spec: {
            containers: [
              {
                image: `registry.kube-system:80/example-image-${process.env.BUILD_ENV}:${process.env.LATEST_COMMIT_ID}`,
                name: "test-deployment-cdk8s",
              },
            ]
          }
        }
      }
    }).toString();
  }
}

const app = new App();
new MyChart(app, 'cdk8s-manifests').toString();
app.synth();
