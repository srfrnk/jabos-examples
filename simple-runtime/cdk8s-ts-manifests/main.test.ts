import { MyChart } from './main';
import { Testing } from 'cdk8s';


describe('Placeholder', () => {
  test('Empty', () => {
    process.env.BUILD_ENV = "DDD";
    process.env.LATEST_COMMIT_ID = "1234";
    const app = Testing.app();
    const chart = new MyChart(app, 'test-chart');
    const results = Testing.synth(chart)
    expect(results).toMatchSnapshot();
  });
});
