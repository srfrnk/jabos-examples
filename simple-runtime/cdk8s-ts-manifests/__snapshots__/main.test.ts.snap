// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Empty 1`] = `
Array [
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "test-deployment-cdk8s",
      },
      "name": "test-deployment-cdk8s",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "test-deployment-cdk8s",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "test-deployment-cdk8s",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "image": "registry.kube-system:80/example-image-DDD:1234",
              "name": "test-deployment-cdk8s",
            },
          ],
        },
      },
    },
  },
]
`;
